hollow/src/test/java/com/netflix/hollow/api/consumer/metrics/AbstractRefreshMetricsListenerTest.java

```diff
@@ -209,4 +214,77 @@ public void refreshEndMetricsReporting(ConsumerRefreshMetrics refreshMetrics) {
         failureTestRefreshMetricsListener.refreshFailed(TEST_VERSION_LOW-1, TEST_VERSION_LOW, TEST_VERSION_HIGH, null);
 
     }
+
+    @Test
+    public void testCycleStart() {  // also exercises reverse delta transition
+        InMemoryBlobStore blobStore = new InMemoryBlobStore();
+        HollowInMemoryBlobStager blobStager = new HollowInMemoryBlobStager();
+        HollowProducer p = HollowProducer
+                .withPublisher(blobStore)
+                .withBlobStager(blobStager)
+                .build();
+        p.initializeDataModel(String.class);
+
+        long version1 = p.runCycle(ws -> {
+            // override cycle start time with a strictly incrementing count to prevent potential
+            // clock skew issue from use of System.currentTimeMillis() that would make test brittle
+            ws.getStateEngine().addHeaderTag(HEADER_TAG_METRIC_CYCLE_START, "1");
+            ws.add("A");
+        });
+
+        class TestRefreshMetricsListener extends AbstractRefreshMetricsListener {
+            int run = 0;
+
+            @Override
+            public void refreshEndMetricsReporting(ConsumerRefreshMetrics refreshMetrics) {
+                run ++;
+                assertNotNull(refreshMetrics.getCycleStartTimestamp());
+                switch (run) {
+                    case 1:
+                        assertEquals(1L, refreshMetrics.getCycleStartTimestamp().getAsLong());
+                        break;
+                    case 2:
+                    case 5:
+                        assertEquals(2L, refreshMetrics.getCycleStartTimestamp().getAsLong());
+                        break;
+                    case 3:
+                        assertEquals(1L, refreshMetrics.getCycleStartTimestamp().getAsLong());
+                        break;
+                    case 4:
+                        assertEquals(3L, refreshMetrics.getCycleStartTimestamp().getAsLong());
+                        break;
+                }
+            }
+        }
```

Comment:
this is intersting, I don't have a better way to do it :), but, took me a sec to wrap my head around it.